fuel-p2p/src/service.rs

```diff
@@ -108,23 +114,63 @@ impl FuelP2PService {
                 SwarmEvent::NewListenAddr { address, .. } => {
                     return FuelP2PEvent::NewListenAddr(address)
                 }
-                _ => {
-                    // todo: handle other necessary events
-                }
+                _ => {}
             }
         }
     }
+
+    /// Sends RequestMessage to a peer
+    /// If the peer is not defined it will pick one at random
+    pub fn send_request_msg(
+        &mut self,
+        peer_id: Option<PeerId>,
+        message_request: RequestMessage,
+        tx_channel: oneshot::Sender<Result<ResponseMessage, ReqResNetworkError>>,
+    ) -> Result<RequestId, RequestError> {
+        let peer_id = match peer_id {
```

Comment:
Would it make sense to extra this peer selector into a separate function? Perhaps even under an overridable trait?

Reply:
The peer selection will be implemented in PeerManager, until then we can leave it as it is, or I can extract it into an overridable trait / function, let me know what you would prefer.

Reply:
functionality wise it's already easily overridable since the peer id is passed in as a param. A separate function would just be tidier in case we wanted to quickly identify or tweak areas of the code involving peer selection, but this is just a minor nit.